rules/web/proxy_generic/proxy_webdav_search_ms.yml

```
@@ -0,0 +1,38 @@
+title: Search-ms and WebDAV Suspicious Indicators in URL
+id: 5039f3d2-406a-4c1a-9350-7a5a85dc84c2
+status: experimental
+description: Detects URL pattern used by search(-ms)/WebDAV initial access campaigns.
+references:
+    - https://www.trellix.com/en-us/about/newsroom/stories/research/beyond-file-search-a-novel-method.html
+    - https://micahbabinski.medium.com/search-ms-webdav-and-chill-99c5b23ac462
+author: Micah Babinski
+date: 2023/08/21
+tags:
+    - attack.initial_access
+    - attack.t1584
+    - attack.t1566
+logsource:
+    category: proxy
+detection:
+    selection_search_ms:
+        c-uri|contains|all:
+            - 'search' # Matches on search:query= or search-ms:query=
+            - ':query='
+            - 'webdav'
+    selection_search_term:
+        c-uri|contains:
+            # Note: Add additional keywords for additional coverage
+            - 'agreement'
+            - 'invoice'
+            - 'notice'
+            - 'payment'
+    filter_main_local_ips:
+        dst_ip|cidr:
+            - '127.0.0.0/8'
+            - '10.0.0.0/8'
+            - '172.16.0.0/12'
+            - '192.168.0.0/16'
```

```suggestion
    filter_main_local_ips:
        dst_ip|cidr:
            - '127.0.0.0/8'
            - '10.0.0.0/8'
            - '172.16.0.0/12'
            - '192.168.0.0/16'
            - '::1'  # IPv6 loopback
            - 'fe80::/8'  # IPv6 link-local addesses
            - 'fc00::/7'  # IPv6 private addresses
```
First time use of the cidr modifier for IPv6 addresses. Let's see what brakes :)